### ISSUE TITLE
Age encrypted files not automatically decrypting

### ISSUE THREAD

**Issue by FlexibleToast** [signageos/vscode-sops#52]
I've tried setting just about every setting and I cannot get the automatic decryption to work. Their must be something I'm missing. I've added the `SOPS_AGE_KEY_FILE` and `SOPS_AGE_KEY` to my `~/.bashrc`

```bash
# Environment Variables
export SOPS_AGE_KEY_FILE=$HOME/.config/sops/age/keys.txt
export SOPS_AGE_KEY=age1sys8sskf58gzct4f433lnd5lke354netv43sc4s0swvkjdd5j53spz9xm7
```

 I've created a `.sops.yaml`

```yaml
creation_rules:
    - age: >-
          age1sys8sskf58gzct4f433lnd5lke354netv43sc4s0swvkjdd5j53spz9xm7
```

I've even created a `.sopsrc`

```
ageKeyFile: $HOME/.config/sops/age/keys.txt
```
I've tried pointing the extension to the key using the absolute path in the extension settings... Yet still when I open the encrypted .env file I have, it stays encrypted. 

I know sops works because I can use `sops --encrypt --in-place file.env` to encrypt and `sops --decrypt --in-place file.env` to decrypt it.

**Comment by vterdunov** [comment]
I guess the extension doesn't work with dotenv files if their names other than `.env`

**Comment by misak113** [comment]
Yes. That can be the case. Or maybe you can alternatively set the current file mimeType (format) in the bottom bar "Select language mode".

**Comment by FlexibleToast** [comment]
I'm not sure I understand. The files end in `.env`. To try out a literal `.env` I copied one of my encrypted files and named it `.env` and it still isn't decrypting automatically. And yes, VScode is recognizing all the files as 'Enviroment Variables' type.

**Comment by FlexibleToast** [comment]
To be even more thorough I tried with a yaml and an ini file. Worked great manually via the cli, but it didn't get decrypted automatically in VScode.

**Comment by misak113** [comment]
Ok, I understand your problem. I already trying to figure it out earlier. The problem is, that the extension only listening on specific file types (mimeTypes) of VSCode. Those are following: `'yaml', 'json', 'ini', 'dotenv', 'plaintext'`.

The file type you are looking for is `dotenv`. However your VSCode currently cannot recognize (associate) the file as `dotenv`. Maybe you use different extension for "dotenv" files. Here is what I use and recommend:
> Name: DotENV
> Id: mikestead.dotenv
> Description: Support for dotenv file syntax
> Version: 1.0.1
> Publisher: mikestead
> VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=mikestead.dotenv

I've just tried the extension in the VSCode and it did work. Only what I have to do is to associate the ".env" file-extension files with the `mikestead.dotenv` vscode extension. So it recognize the file when is open immediately (not after).

Association the file ext with file format:
1. Click bottom button with current file format (Properties/Environment Variables/Unknown, etc.)
2. Configure File Association for '.env'.
3. Select the "Environment Variables (dotenv)" (the braced dotenv is important).
4. Close the file and open it again.

If you have any recommendation how it can be done better way, please let me know and I'll try to implemented that.

**Comment by FlexibleToast** [comment]
I'm on vacation now, but I can try that when I return. I'm not confident that is the issue though as I've tested with a `yaml` file before and it also didn't auto decrypt.

**Comment by FlexibleToast** [comment]
It turns out I already have that extension installed. After clicking "Configure File Association for '.env'", I'm given a list of filetypes and none of them have any braced anything. However, If I just look at the drop down that comes up after clicking the bottom current file format, I can see that `Environment Variables (dotenv)` is already the configured language.

**Comment by fisherbln** [comment]
I'm having this exact same problem with with VS Code version 1.74.3 Universal for Mac. I also tried the M1 native version. SOPS just doesn't work in this version. If I downgrade to 1.72.2, it works fine. Been this way ever since 1.73.x

**Comment by electropolis** [comment]
Got the same problem. But I din't configure `.sopsrc` because I already have `.sops.yaml` with content

```yaml
---

keys:
  - &konrad age1qh8qcjnxdy26t4kdc7wn4sxwx2npl4fus3me06pva2wxyc0hpqzqgr9xad

# creation rules are evaluated sequentially, the first match wins
creation_rules:
  - path_regex: .*secrets.yaml$
    key_groups:
      - age:
          - *konrad
```

and `SOPS_AGE_KEY_FILE` env added. Command `sops` work but VSC is doing nothing. Can't say why. I think there is lack of information in README

**Comment by FlexibleToast** [comment]
Renaming to `secret.env` did not work for me.
